rust/src/tests/chmod.rs

```diff
@@ -101,3 +107,77 @@ fn clear_isgid_bit(ctx: &mut SerializedTestContext) {
     assert_eq!(actual_mode & 0o7777, expected_mode.bits());
     //TODO: FreeBSD "S_ISGID should be removed and chmod(2) should success and FreeBSD returns EPERM."
 }
+
+crate::test_case! {
```

Comment:
This sticky stuff doesn't have anything to do with ENOTDIR, right?  It should go in a separate PR.

Reply:
Yes, sorry!